spec/classes/foreman_proxy__proxydhcp__spec.rb

```diff
@@ -194,6 +194,60 @@
             'interfaces' => ['eth0']
         ) end
       end
+
+      context "as manager of ACLs for dhcp" do
+        let :facts do
+          facts.merge({:ipaddress_eth0 => '192.168.100.20',
+                       :ipaddress      => '192.168.100.20',
+                       :netmask_eth0   => '255.255.255.0',
+                       :network_eth0   => '192.168.100.0'})
+        end
+
+        let :pre_condition do
+          "class {'foreman_proxy':
+            dhcp_manage_acls  => true,
+          }"
+        end
+
+        it do should contain_exec('setfacl_etc_dhcp').
+          with_command("setfacl -R -m u:foreman-proxy:rx /etc/dhcp")
+        end
+
+        it do should contain_exec('setfacl_var_lib_dhcp').
+          with_command("setfacl -R -m u:foreman-proxy:rx /var/lib/dhcpd")
+        end
+      end
+
+      context "as manager of ACLs for dhcp for RedHat only by default" do
```

Comment:
Here I'm unsure. On the one hand you only want a test to be as specific as possible but we've noticed that compiling the entire class takes quite a bit of time so moving the assertions in the ```without parameters``` context has a slight preference.